#Author: Arunim Joarder

comm_config:
  baudrate: 57600
  biped_port: '/dev/ttyUSB0'
  torso_port: '/dev/ttyUSB1'

general:
  behave_mode: 'init'
  switch_index: 0
  
  prf_vel: 0
  prf_acc: 0
  max_vel: 0

  p_gain_pos: 0
  i_gain_pos: 0
  d_gain_pos: 0

  moving_vel: 0

  biped_dxls: [[1, 2, 0],                 # info about dynamixels in biped
               [1, 2, 0],                     # ith array = [id, protocol, initial_angle]
               [1, 2, 0],
               [1, 2, 0],
               [1, 2, 0],
               [1, 2, 0],
               [1, 2, 0],
               [1, 2, 0],
               [1, 2, 0],
               [1, 2, 0],
               [1, 2, 0],
               [1, 2, 0]]
  
  torso_dxls: [[1, 1, 0],                 # info about dynamixels in biped
               [1, 1, 0],                     # ith array = [id, protocol, initial_angle]
               [1, 1, 0],
               [1, 1, 0],
               [1, 1, 0],
               [1, 1, 0],
               [1, 1, 0]]

dynamixel_config:
  MX_2:                                 # Control table addresses and lengths for MX series for PROTOCOL 2.0
    torque_enable:  [64 , 1]                # data_name:  [data_addr, data_size]
    goal_pos:       [116, 4]
    present_pos:    [132, 4]
    prf_vel:        [112, 4]
    prf_acc:        [108, 4]
    drive_mode:     [10 , 1]
    moving:         [122, 1]
    max_vel:        [44 , 4]
    p_gain_pos:     [84 , 2]
    i_gain_pos:     [82 , 2]
    d_gain_pos:     [80 , 2]
    hardware_error: [70 , 1]

  AX_1:                                 # Control table addresses and lengths for AX series for PROTOCOL 1.0
    torque_enable:  [24 , 1]                # data_name:  [data_addr, data_size]
    goal_pos:       [30 , 4]
    present_pos:    [36 , 4]
    moving:         [46 , 1]
    moving_vel:     [32 , 2]

modes:  
  init:                           # switch_index = 0    (initial position - fully straight)
    info: [[0, 0], [0, 0]]
  stand:                          # switch_index = 1    (stand position - knees bent)
    info: [[0, 0], [0, 0]]
  walk:                           # switch_index = 2    (walk motion)
    index: 0
    swing_leg: 'left'
    info: [[1, 2, 3], [4, 5, 6]]
  turn:                           # switch_index = 3    (turn motion)
    dir:  'left'
    info: [[0, 0], [0, 0]]
  rise:                           # switch_index = 4    (rise motion)
    index: 0
    info: [[0, 0], [0, 0]]